helpers/pc-bios.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Lightweight ACPI Implementation
+ * Copyright (C) 2018-2019 the lai authors
+ */
+
+#include <acpispec/tables.h>
+#include <lai/core.h>
+#include <lai/helpers/pc-bios.h>
+
+static uint8_t lai_bios_calc_checksum(void *ptr, size_t size) {
+	uint8_t sum = 0;
+	for (size_t i = 0; i < size; i++)
+		sum += ((uint8_t *)ptr)[i];
+	return sum;
+}
+
+int lai_bios_detect_rsdp_within(uintptr_t base, size_t length,
+		uintptr_t *rsdp_addr, uintptr_t *rsdt_addr) {
+    int version = 0;
+	uint8_t *window = laihost_map(base, length);
+	for (size_t off = 0; off < length; off += 16) {
+		acpi_rsdp_t *rsdp = (acpi_rsdp_t *)(window + off);
+
+		if (memcmp(rsdp->signature, "RSD PTR ", 8))
+			continue;
+		if (lai_bios_calc_checksum(rsdp, sizeof(acpi_rsdp_t)))
+			continue;
+
+        // TODO: Handle XSDTs.
+		if(rsdp_addr)
+			*rsdp_addr = base + off;
+		if(rsdt_addr)
+			*rsdt_addr = rsdp->rsdt;
+		version = 1;
+        goto done;
```

Comment:
Is this goto really necessary, a `break` would seem better

Reply:
This goto is necessary once we add a `laihost_unmap` function (which we should do).